firmware/watchguard_sysa_dl.ksy

```diff
@@ -0,0 +1,137 @@
+meta:
+  id: watchguard_sysa_dl
+  endian: be
+  title: Watchguard Fireware web update (Sysa-dl) file format
+  license: MIT
+
+seq:
+  - id: fw_header
+    type: fw_header
+    size: 24
+
+  - id: sections
+    type: sections
+    size: fw_header.file_size
+
+types:
+  fw_header:
+    seq:
+      - id: md5sum
+        size: 0x10
+
+      - id: file_size
+        type: u4
+
+      - id: magic
+        type: u4
+        enum: magic
+
+  sections:
+    seq:
+      - id: section
+        type: section
+        repeat: eos
+
+  section:
+    seq:
+      - id: head
+        type: head
+
+      - id: data
+        size: head.data_size
+        type:
+          switch-on: head.name.content
+          cases:
+            '"REBOOT"': reboot_data
+            '"info"': info_data
+            '"HMAC"': hmac_data
+            '"WGPKG"': wgpkg_data
+
+
+  name:
+    seq:
+      - id: type2_magic
+        type: u1
+        if: is_type2
+
+      - id: content
+        type: strz
+        size: 'is_type2 ? 0xf : 0x10'
+        encoding: ASCII
+
+    instances:
+      type2:
+        pos: 0
+        type: u1
+
+      is_type2:
+        value: type2 == 0x7f
+
+  head:
+    seq:
+      - id: name
+        type: name
+        size: 0x10
+
+      - id: unknown_data
+        size: 24
+        if: name.is_type2
+
+      - id: data_size
+        type: u4
+
+      - id: md5sum
+        size: 0x10
+        if: name.is_type2
+
+  reboot_data:
+    seq:
+      - id: perm
+        type: encoded_perm
+        size: 0x8
+
+  encoded_perm:
+    seq:
+      - id: unknown_dword1
+        type: u4
+      - id: unknown_dword2
+        type: u4
+
+  info_data:
+    seq:
+      - id: info
+        type: strz
+        encoding: ASCII
+        size-eos: true
+
+  hmac_data:
+    seq:
+      - id: sha1_sign #key: etaonrishdlcupfm
+        size: 20
+
+  wgpkg_data:
+    seq:
+      - id: meta_info
+        type: strz
+        encoding: ASCII
+        terminator: 0
+
+      - id: magic
+        contents: "WGPKG\0"
+
+      - id: unknown_word
+        size: 2
+
+      - id: unknown_dword
+        type: u4
+      - id: data_size
+        type: u4
+
+      - id: compressed_wpkg #bz2
+        #size-eos: true
+        size: data_size
+
+enums:
+  magic:
+    0x1261_1920: type1
+    0x1561_1928: type2
```

Comment:
There has to be a newline at EOF in order to comply with the [style guide](https://doc.kaitai.io/ksy_style_guide.html#general):

> MUST use:
>
> - (...)
> - **Trailing newline character in a .ksy file.**

In this repository, it's recommended to set up your editor to honor the [project `.editorconfig`](https://github.com/kaitai-io/kaitai_struct_formats/blob/1eae5145325bb992e20dae838dd68d04814dda46/.editorconfig#L8) (for example, there is a plugin [_EditorConfig for VS Code_](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)).

---

But I personally recommend configuring your editor/IDE to sanitize whitespace in all projects. For example in VS Code, there is this setting:

> **Files: Insert Final Newline**
> 
> - [x] When enabled, insert a final new line at the end of the file when saving it.

and related (to also ensure exactly one newline at the end):

> **Files: Trim Final Newlines**
> 
> - [x] When enabled, will trim all new lines after the final new line at the end of the file when saving it.

In JetBrains IDEs (although I assume you're not using them, because they [respect EditorConfig](https://editorconfig.org/#pre-installed) out of the box, so in this project you wouldn't be able to save a file without a newline at EOF at all) these settings are in **Editor > General**, section *On Save*:

> - [x] Remove trailing blank lines at the end of saved files
> - [x] Ensure every saved file ends with a line break